detect_secrets_server/storage/file.py

```diff
@@ -0,0 +1,99 @@
+import json
+import os
+
+from .base import BaseStorage
+from .base import get_filepath_safe
+from .base import LocalGitRepository
+
+
+class FileStorage(BaseStorage):
+    """For state management without a database.
+
+    Structure:
+        root
+          |- repos      # This is where git repos are cloned to
+          |- tracked    # This is where meta files containing state reside
+    """
+
+    def setup(self, repo_url):
+        super(FileStorage, self).setup(repo_url)
+
+        storage_root = os.path.join(self.root, 'tracked')
+        if not os.path.isdir(storage_root):
+            os.makedirs(storage_root)
+
+        return self
+
+    def get(self, key):
+        """
+        :raises: FileNotFoundError
+        :raises: ValueError
+        """
+        filename = self.get_tracked_file_location(key)
+        with open(filename) as f:
+            return json.load(f)
+
+    def put(self, key, value):
+        """
+        :raises: ValueError
+        """
+        filename = self.get_tracked_file_location(key)
+        with open(filename, 'w') as f:
+            f.write(json.dumps(value, indent=2, sort_keys=True))
+
+    def get_tracked_file_location(self, key):
+        return get_filepath_safe(
+            os.path.join(self.root, 'tracked'),
+            '{}.json'.format(key),
+        )
+
+    def get_tracked_repositories(self):
+        filepath = get_filepath_safe(
+            self.root,
+            'tracked',
+        )
+
+        for root, _, files in os.walk(filepath):
+            for filename in files:
+                with open(os.path.join(root, filename)) as f:
+                    yield json.loads(f.read()), False
```

Comment:
Nit: Feel like a namedtuple would be good for `get_tracked_repositories`, since you wouldn't have to look elsewhere to know what was False